File: trunk/src/test/org/codehaus/testdox/intellij/AbstractTestElementTest.java
package org.codehaus.testdox.intellij;

import javax.swing.Icon;

import junit.framework.TestCase;

import org.intellij.openapi.testing.MockApplicationManager;

import com.intellij.openapi.actionSystem.Presentation;

import org.codehaus.testdox.intellij.actions.RenameTestAction;

public class AbstractTestElementTest extends TestCase {

    private AbstractTestElement abstractTestElement = new AbstractTestElement() {
        public String displayString() {
            throw new UnsupportedOperationException();
        }

        public Icon icon() {
            throw new UnsupportedOperationException();
        }
    };

    public void testIsAssociatedToANullPsiElementByDefault() {
        MockApplicationManager.reset();
        assertSame(NullPsiElement.INSTANCE, abstractTestElement.psiElement());
    }

    public void testDoesNotJumpToPsiElementByDefault() {
        assertFalse(abstractTestElement.jumpToPsiElement());
    }

    public void testUsesReferenceEqualityToDefineTheDefaultNaturalOrder() {
        assertEquals(-1, abstractTestElement.compareTo(new AbstractTestElement() {
                         public String displayString() {
                             throw new UnsupportedOperationException();
                         }

                         public Icon icon() {
                             throw new UnsupportedOperationException();
                         }
                     }));

        assertEquals(0, abstractTestElement.compareTo(abstractTestElement));
    }

    public void testAlwaysDisablesTheRepresentationOfAnActionWhenAskedToUpdateIt() {
        Presentation presentation = new RenameTestAction().getTemplatePresentation();
        presentation.setEnabled(true);

        abstractTestElement.updatePresentation(presentation);
        assertFalse("action representation should have been disabled", presentation.isEnabled());
    }

    public void testDoesNothingWhenAskedToRenameTheUnderlyingTestedClass() {
        abstractTestElement.rename(null);
    }

    public void testDoesNothingWhenAskedToDeleteTheUnderlyingTestedClass() {
        abstractTestElement.delete(null);
    }
}

File: trunk/test/scala/org/codehaus/testdox/intellij/AbstractTestElementTest.scala
package org.codehaus.testdox.intellij

import junit.framework.Assert._
import junit.framework.TestCase
import org.codehaus.testdox.intellij.actions.RenameTestAction
import org.intellij.openapi.testing.MockApplicationManager
import javax.swing.Icon

class AbstractTestElementTest extends TestCase {

  private val abstractTestElement = new AbstractTestElement() {
    override val displayString = "dummy"
    override val icon: Icon = null
  }

  def testIsAssociatedToANullPsiElementByDefault() {
    MockApplicationManager.reset()
    assertSame(NullPsiElement.INSTANCE, abstractTestElement.psiElement)
  }

  def testDoesNotJumpToPsiElementByDefault() {
    assertFalse(abstractTestElement.jumpToPsiElement())
  }

  def testUsesReferenceEqualityToDefineTheDefaultNaturalOrder() {
    assertEquals(-1, abstractTestElement.compareTo(new AbstractTestElement() {
      override val displayString = "dummy"
      override val icon: Icon = null
    }))

    assertEquals(0, abstractTestElement.compareTo(abstractTestElement))
  }

  def testAlwaysDisablesTheRepresentationOfAnActionWhenAskedToUpdateIt() {
    val presentation = new RenameTestAction().getTemplatePresentation()
    presentation.setEnabled(true)

    abstractTestElement.update(presentation)
    assertFalse("action representation should have been disabled", presentation.isEnabled())
  }

  def testDoesNothingWhenAskedToRenameTheUnderlyingTestedClass() {
    abstractTestElement.rename(null)
  }

  def testDoesNothingWhenAskedToDeleteTheUnderlyingTestedClass() {
    abstractTestElement.delete(null)
  }
}
